### DISCUSSION TITLE
Code block syntax highlighting using built-in functions

### DISCUSSION THREAD

**Discussion by igor-baiborodine** [Lednerb/bilberry-hugo-theme#362] [Ideas]
Related to [issue #359](https://github.com/Lednerb/bilberry-hugo-theme/issues/359). 

Currently, the `highlight.js` library is used for syntax highlighting. Unfortunately, this library has no support for displaying line numbers.
As an alternative, `highlight.js` can be replaced by the syntax highlighting built-in functions provided by the Goldmark markdown parser.
https://gohugo.io/content-management/syntax-highlighting/

**Comment by Lednerb** [comment]
Another approach can maybe be this interesting project: https://torchlight.dev

The focus and diff functions seem to be very nice.
However there will be a big overhead implementing it with hugo...

**Comment by jrpetersjr** [comment]
It's almost been a year and nothing has been decided on this. torchlight.dev needs all new setup and highlight.js doesn't support everything needed such as line number. 

Hugo has built in highlighting that runs in Go (Fast) and runs at the time of creation instead of on the browser like highlight.js. It's also would be the most idiomatic way of dealing with it as it's built into the SSG that we are using. Would now be the right time to remove that highlight.js from the project?

**Reply by jrpetersjr** [reply]
I understand that highlight.js can be removed, but I propose that highlight.js should be removed entirely or at least set to not be included as default.

**Reply by igor-baiborodine** [reply]
@jrpetersjr, the code highlighting issue will be addressed in the upcoming `v4` version, which will be in development starting in March.
